File: C/pset2/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>

float get_letters();
float get_words();
float get_sentences();

int main(void)
{
    string sentence = get_string("Text: ");
    int n = strlen(sentence);
    float w = 100 / get_words(sentence, n);
    float l = get_letters(sentence, n) * w;
    float s = get_sentences(sentence, n) * w;
    int index = (0.0588 * l - 0.296 * s - 15.8) + 0.5;
    if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %d\n", index);
    }
}

float get_letters(string sentence, int length)
{
    float letters = 0;
    for (int i = 0; i < length; i++)
    {
        if ((sentence[i] > 64 && sentence [i] < 91) || (sentence[i] > 96 && sentence[i] < 123))
        {
            letters++;
        }
    }
    return letters;
}

float get_words(string sentence, int length)
{
    int words = 1;
    for (int i = 0; i < length; i++)
    {
        if (sentence[i] == ' ')
        {
            words++;
        }
    }
    return words;
}

float get_sentences(string sentence, int length)
{
    int sentences = 0;
    for (int i = 0; i < length; i++)
    {
        if (sentence[i] == '.' || sentence[i] == '!' || sentence[i] == '?')
        {
            sentences++;
        }
    }
    return sentences;
}

File: Python/pset6/readability.py
def main():
    text = input("Text: ")
    w = words(text)
    l = letters(text) * (100 / w)
    s = sentences(text) * (100 / w)
    index = int(0.0588 * l - 0.296 * s - 15.8 + 0.5)

    if index < 1:
        print("Before Grade 1")

    elif index > 16:
        print("Grade 16+")

    else:
        print(f"Grade {index}")


def letters(text):
    letters = 0
    for i in range(len(text)):
        if text[i].isalpha():
            letters += 1
    return letters


def words(text):
    words = 1
    for i in range(len(text)):
        if text[i] == " ":
            words += 1
    return words


def sentences(text):
    sentences = 0
    for i in range(len(text)):
        if text[i] == "." or text[i] == "!" or text[i] == "?":
            sentences += 1
    return sentences


main()
